src/Compilers/CSharp/Portable/Symbols/Compilation_WellKnownMembers.cs

```diff
@@ -75,6 +75,25 @@ internal Symbol GetWellKnownTypeMember(WellKnownMember member)
             return _lazyWellKnownTypeMembers[(int)member];
         }
 
+        /// <summary>
+        /// Lookup well-known member declaration in provided type.
+        ///
+        /// If a well-known member of a generic type instantiation is needed use this method to get the corresponding generic definition and
+        /// <see cref="MethodSymbol.AsMember"/> to construct an instantiation.
+        /// </summary>
+        /// <param name="type">Type that we'll try to find member in.</param>
+        /// <param name="relativeMember">A reference to a well-known member type descriptor. Note however that the type in that descriptor is ignored here.</param>
+        internal Symbol GetWellKnownTypeMember(NamedTypeSymbol type, WellKnownMember relativeMember)
```

Comment:
If I make it static, I will have to make my own signature comparer or pass one around (which still requires a Compilation to construct). Why not keep it as instance method on Compilation?